fabric-data-generation-api-v1/src/main/java/net/fabricmc/fabric/api/datagen/v1/JsonKeySortOrderAdder.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.datagen.v1;
+
+/**
+ * Provides a way to add keys and priorities for sorting them in generated files.
+ */
+@FunctionalInterface
+public interface JsonKeySortOrderAdder {
+	/**
+	 * Adds a key with a priority for sorting.
+	 * @param key The key to compare.
+	 * @param priority The priority for this key. Lower numbers appear before ones with higher numbers. The default priority is 2.
```

Comment:
The default priority here doesn't apply to `type` (`0`) or `parent` (`1`). Should these exceptions that are default in vanilla be documented as well?

Reply:
That would be useful for an additional comparison. Should the two keys provided by fabric (`fabric:load_conditions` and `fabric:type`) also be included then?

Reply:
I'd keep them as implementation details.

Reply:
So do I not add them at all to the `@implNote` or just the vanilla ones?

Reply:
Vanilla ones are fine but could also get outdated. Maybe we just add `@see <field reference>`?

Reply:
Alright, I added a reference to `net.minecraft.data.DataProvider#JSON_KEY_SORT_ORDER`.